pkg/apply/merge.go

```diff
@@ -150,8 +146,9 @@ func mergeLabels(current, updated *uns.Unstructured) {
 	for k, v := range updatedLabels {
 		curLabels[k] = v
 	}
-
-	updated.SetLabels(curLabels)
+	if len(curLabels) > 1 {
```

Comment:
Will this block if from setting labels to empty?

Reply:
The original logic is a merging instead of replacing. So the controller can only add new labels or do nothing.